framework/include/utils/MooseRandom.h

```diff
@@ -167,6 +167,11 @@ class MooseRandom
       it->second.first = it->second.second;
   }
 
+  LIBMESH_BEST_UNORDERED_MAP<unsigned int, std::pair<mt_state, mt_state> > getStates() { return _states; }
```

Comment:
No. You have not implemented a helper for `LIBMESH_BEST_UNORDERED_MAP`, have you? This only works if this symbol is set to `std::unordered_map`.